includes/gateway/class-omise-payment-installment.php

```diff
@@ -69,27 +71,15 @@ public function init_form_fields() {
 		 * @inheritdoc
 		 */
 		public function payment_fields() {
-			$provider_names = array(
-				'installment_bay'          => __( 'Krungsri', 'omise' ),
-				'installment_first_choice' => __( 'Krungsri First Choice', 'omise' ),
-				'installment_kbank'        => __( 'Kasikorn Bank', 'omise' ),
-				'installment_bbl'          => __( 'Bangkok Bank', 'omise' ),
-				'installment_ktc'          => __( 'Krungthai Card (KTC)', 'omise' ),
-			);
-
-			$currency             = get_woocommerce_currency();
-			$cart_total           = WC()->cart->total;
-			$capabilities         = Omise_Capabilities::retrieve();
-			$installment_backends = $capabilities->getInstallmentBackends( $currency, $this->format_amount_subunit( $cart_total, $currency ) );
-
-			foreach ( $installment_backends as &$backend ) {
-				$backend->provider_code = str_replace( 'installment_', '', $backend->_id );
-				$backend->provider_name = isset( $provider_names[ $backend->_id ] ) ? $provider_names[ $backend->_id ] : strtoupper( $backend->provider_code );
-			}
+			$currency   = get_woocommerce_currency();
```

Comment:
Curious if this is the correct way? I would expect this to get the general currency setting for the site. In my experience with ecommerce sites, this isn't always correct as sometimes multi-currency plugins are in play. Is the a currency actually stored in the cart - it would probably be a better one to use?

Reply:
I was looking through the whole code of WooCommerce and seems they heavily used `get_woocommerce_currency()` in everywhere. But good point on the multi-currency thing.
I'll do a bit of research and get back to you soon 👍

Reply:
@jonrandy Checked. I'm convinced that this is the right way (for now), as for 
1. As WooCommerce leaves the hook here, I believe that this is an intention of WC to let those multi-currency feature will hook this filter in order to alternate the currency code.
```php
function get_woocommerce_currency() {
	return apply_filters( 'woocommerce_currency', get_option( 'woocommerce_currency' ) );
}
```

2. I checked how WC formats and displays its price at the checkout page, found that `get_woocommerce_currency()` has been heavily used everywhere.

3. Order has not been created yet at this point. We cannot use a currency from an order object (it will be created at the 'place order' step (aka. payment processing step)).